src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java

```diff
@@ -710,14 +710,20 @@ public Response deleteDataverseLinkingDataverse(@Context ContainerRequestContext
     @GET
     @AuthRequired
     @Path("{identifier}/metadatablocks")
-    public Response listMetadataBlocks(@Context ContainerRequestContext crc, @PathParam("identifier") String dvIdtf) {
+    public Response listMetadataBlocks(@Context ContainerRequestContext crc,
+                                       @PathParam("identifier") String dvIdtf,
+                                       @QueryParam("onlyDisplayedOnCreate") boolean onlyDisplayedOnCreate,
+                                       @QueryParam("returnDatasetFieldTypes") boolean returnDatasetFieldTypes) {
         try {
-            JsonArrayBuilder arr = Json.createArrayBuilder();
-            final List<MetadataBlock> blocks = execCommand(new ListMetadataBlocksCommand(createDataverseRequest(getRequestUser(crc)), findDataverseOrDie(dvIdtf)));
-            for (MetadataBlock mdb : blocks) {
-                arr.add(brief.json(mdb));
-            }
-            return ok(arr);
+            final List<MetadataBlock> metadataBlocks = execCommand(
+                    new ListMetadataBlocksCommand(
+                            createDataverseRequest(getRequestUser(crc)),
+                            findDataverseOrDie(dvIdtf),
+                            onlyDisplayedOnCreate,
+                            metadataBlockSvc
```

Comment:
Rather than passing in a service, the MetadataBlockServiceBean should probably get added to the [CommandContext class](src/main/java/edu/harvard/iq/dataverse/engine/command/CommandContext.java) - that's the standard pattern for accessing services in commands.